ci/setup_thrift.sh

```diff
@@ -0,0 +1,42 @@
+#!/bin/bash
+
+set -e
+export DEBIAN_FRONTEND=noninteractive
+export THRIFT_VERSION=0.14.1
+
+if ! type cmake > /dev/null; then
+    #cmake not installed, exiting
+    exit 1
+fi
+export BUILD_DIR=/tmp/
+export INSTALL_DIR=/usr/local/
+
+apt install -y --no-install-recommends \
+      libboost-all-dev \
+      libevent-dev \
+      libssl-dev \
+      ninja-build
+
+pushd $BUILD_DIR
+wget https://github.com/apache/thrift/archive/refs/tags/v${THRIFT_VERSION}.tar.gz
+tar -zxvf v${THRIFT_VERSION}.tar.gz
+cd thrift-${THRIFT_VERSION}
+mkdir -p out
+pushd out
+cmake -G Ninja .. \
+    -DBUILD_COMPILER=OFF \
+    -DBUILD_CPP=ON \
+    -DBUILD_LIBRARIES=ON \
+    -DBUILD_NODEJS=OFF \
```

Comment:
All the options turned off here are turned on by default in thrift repo. They are not used by our C++ build so turn them off explicitly would introducing more dependence to our CI build.

https://github.com/apache/thrift/blob/70992f1e74e525461121fb9e607000b19f31a4ca/build/cmake/DefineOptions.cmake#L115